notebooks/07_nsight_compute.ipynb

```diff
@@ -0,0 +1,613 @@
+    "\n",
+    "The Thicket `add_ncu` function takes one required argument and one optional arguement. The required argument `ncu_report_mapping` is the mapping from the NCU report file to the corresponding calltree profile run. The optional argument `chosen_metrics` allows for a subselection of the NCU performance metrics to add, since there can be hundreds of NCU performance metrics. By default we add all metrics."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "60ff359c",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Add NCU to thicket\n",
+    "ncu_metrics = [\n",
+    "    \"gpu__time_duration.sum\",\n",
+    "    \"sm__throughput.avg.pct_of_peak_sustained_elapsed\",\n",
+    "    \"smsp__maximum_warps_avg_per_active_cycle\",\n",
+    "]\n",
+    "# Add in metrics\n",
+    "tk_cap.add_ncu(\n",
+    "    ncu_report_mapping=ncu_report_mapping, \n",
+    "    chosen_metrics=ncu_metrics,\n",
+    ")\n",
+    "tk_cap.dataframe.head(20)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "d69ba22d",
+   "metadata": {},
+   "source": [
+    "## 5. Visualize the NCU Performance Data on the Calltree"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "877c1044",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "print(tk_cap.tree(\n",
+    "    metric_column=\"sm__throughput.avg.pct_of_peak_sustained_elapsed\",\n",
+    "    expand_name=True,\n",
+    "    ))"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "b2560242",
+   "metadata": {},
+   "source": [
+    "## 6. Create Instruction Roofline Plots\n",
+    "\n",
+    "We can make roofline plots using the metrics we have collected using Nsight Compute. The Roofline sets an upper bound on performance of a kernel depending on its operational intensity. We will use roofline plots to understand the performance of RAJAPerf kernels.\n",
+    "\n",
+    "Instruction Roofline Models were introduced by Ding et. al to better characterize GPU workloads by looking at instruction intensity. We walk through the creation of instruction roofline models for the Appications group of the RAJAPerf kernels below. \n",
+    "\n",
+    "More references on methodology for roofline models:\n",
+    "\n",
+    "* Samuel Williams, Andrew Waterman, and David Patterson. 2009. Roofline: an insightful visual performance model for multicore architectures. Commun. ACM 52, 4 (April 2009), 65–76. https://doi.org/10.1145/1498765.1498785\n",
+    "\n",
+    "* Nan Ding, Muaaz Awan, Samuel Williams. 2022. Instruction Roofline: An insightful visual performance model for GPUs. Concurrency and Computation: Practice and Experience\n",
+    "https://doi.org/10.1002/cpe.6591"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "d5073924",
+   "metadata": {},
+   "source": [
+    "Step 1 is to create a Thicket using a CUDA Activity Profile, and add the NCU data. We use `RAJA_CUDA` variant data of the `block_256` tuning."
```

Comment:
@michaelmckinsey1 can we add the actual command line that used to generate the ncu files? May need to ask @Yejashi - it was a specific group to get these metrics. Or at the very least let's list the counters we need.

Reply:
[093bb9b](https://github.com/LLNL/thicket-tutorial/pull/46/commits/093bb9b76cbc7e690163d6763af73a8eda53b18d)